lib/elixir/lib/list.ex

```diff
@@ -591,7 +591,12 @@ defmodule List do
   """
   @spec to_string(:unicode.charlist) :: String.t
   def to_string(list) when is_list(list) do
-    case :unicode.characters_to_binary(list) do
+    try do
+       :unicode.characters_to_binary(list)
+    rescue
+      ArgumentError ->
+        raise ArgumentError, "cannot make this list into a string (use 'inspect(list)' for its internal representation as a string)"
```

Comment:
@josevalim I like that. Is the convention lower-case first letter even if there are multiple sentences?

Do you think mentioning "inspect" is a bad idea? This error message is far less cryptic than the one we're replacing, but I still think newbies may be confused because they don't realize that to_string is a strict/implicit conversion (in Ruby parlance), and inspect is the way to go if you want something looser. Would love to make that fact more discoverable somehow.